src/polyswarm_api/__init__.py

```diff
@@ -735,58 +736,110 @@ def _reveal_closed(self, result):
         """
         return await self._delete_hunt(hunt_id, 'historical')
 
-    async def _get_hunt_results(self, hunt_id=None, scan_type='live'):
+    async def _get_hunt_results(self, hunt_id=None, scan_type='live', limit=MAX_HUNT_RESULTS, offset=0,
+                                all_results=False, with_metadata=False, with_bounties=False):
         """
 
         :param hunt_id: Rule ID (None if latest rule results are desired)
         :param scan_type: Type of scan, "live" or "historical"
+        :param limit: Limit the number of scan results, returns the most recent hits
+        :param offset: Offset into the result set to return
+        :param all_results: Boolean on whether to fetch all results. Note: this ignores limit/offset and can take awhile.
+        :param with_metadata: Whether to include metadata in the results
+        :param with_bounties: Whether to include bounty results in the results
         :return: Matches to the rules
         """
 
-        params = {}
+        if limit > MAX_HUNT_RESULTS:
+            limit = MAX_HUNT_RESULTS
+
+        # ignore offset/limit results in case we want all results
+        if all_results:
+            limit = MAX_HUNT_RESULTS
+            offset = 0
+
+        params = {
+            'limit': limit,
+            'offset': offset,
+        }
+
         if hunt_id is not None:
             params['id'] = hunt_id
 
+        if with_bounties:
+            params['with_bounty_results'] = 'true'
+
+        if with_metadata:
+            params['with_metadata'] = 'true'
+
         async with self.get_semaphore:
             logger.debug('Reading results with api-key %s', self.api_key)
             async with aiohttp.ClientSession() as session:
                 try:
-                    async with session.get('{hunt_uri}/{scan_type}/results'.format(hunt_uri=self.hunt_uri,
+                    async def _make_request(hunt_uri, params):
+                        async with session.get('{hunt_uri}/{scan_type}/results'.format(hunt_uri=hunt_uri,
```

Comment:
Should the async semaphore be here to avoid hammering the api too hard?

Reply:
The semaphore is there to prevent multiple outstanding connections; I believe this code block will only grab it once, but I agree it is cleaner to grab it right before making the request.